hazelcast/src/main/java/com/hazelcast/client/impl/spi/impl/ClientClusterServiceImpl.java

```diff
@@ -99,7 +99,10 @@ public Cluster getCluster() {
     }
 
     public boolean isEnterprise() {
-        return this.getMemberList().size() > 0 && this.isFailoverSupported;
+        if (this.getMemberList().isEmpty()) {
```

Comment:
This check does not seem to be correct. The member list may not be cleared during CLIENT_DISCONNECTED state. You should add lifecycleListener and keep an internal state info and check that flag.